packages/kit-headless/src/components/checkbox/checkbox.tsx

```diff
@@ -10,7 +11,7 @@ export type LabelProps = QwikIntrinsicElements['label'] & {
   htmlFor?: string;
 };
 
-export const Label = component$(({ ...props }: LabelProps) => {
+export const Label: Component<LabelProps> = component$(({ ...props }: LabelProps) => {
```

Comment:
isn't this type being automatically inferred from the `component$` function? 
Do we need to add it to solve the issue? or is it from a different reason?

Reply:
It should be. It is not inferred - and I believe this has to do with [this TS bug](https://github.com/microsoft/TypeScript/issues/47663)

Reply:
The problem is a **TS2742** error which results in a **missing `combobox.d.ts`** file. 

It seems you are worried that every component would need to be type annotated and that's verbose and unnecessary since the type is inferred. 

I 100% agree. 


So why add this? 
- it seems the TS error (https://github.com/microsoft/TypeScript/issues/47663) appears for pnpm reports (like this one)
- switching to yarn or npm is another option - that will make this unnecessary (need to be tested)
- this is not required for every component - the ones in this PR are all that had this error

Context: I should clarify that I tested using the `core.d.ts` that will be generated after the [qwik issue](https://github.com/BuilderIO/qwik/issues/5140) is _merged and released_  using a fork of qwik.